### ISSUE TITLE
Avoid classifier re-training

### ISSUE THREAD

**Issue by a-solovyev12** [pcahan1/singleCellNet#37]
Dear authors, 

Thank you for developing singleCellNet. 
I have a question regarding the common gene set used for training. Suppose I am training a classifier that will be used to predict multiple different datasets automatically - I will not know in advance which genes will be present there and which absent. Is there a way to train a generic model that will work on different datasets or will I need to train a new one for each combination of reference/query datasets? Thanks!

**Comment by yuqiyuqitan** [comment]
So I would recommend finding the intersected genes between your training data and your multiple query datasets if you have all of them already to train the generic classifier. This way the classification results can be compared across query studies. However, you can also just train a classifier for each query dataset. Sometimes it gives better performance.

**Comment by a-solovyev12** [comment]
Thanks for your response @yuqiyuqitan. 
I'm afraid we don't know in advance which data will be used as query. The aim is to build a library of models pre-trained on a range of labelled datasets, which will predict the unseen data provided by others. So, the information about genes will only be available at runtime, and we don't want to re-train the model every time prediction is made. Will it work for this use case?

**Comment by yuqiyuqitan** [comment]
As the tool is built currently, you don't need to retrain the model every time a new prediction is made as long as the feature genes are shared. but I don't think it is possible to just have one classifier that will work for all datasets that might come through the door.

**Comment by pcahan1** [comment]
One possibility to get around this problem would be to impute either expression for genes missing in the query data, or impute gene pair values. A totally naive approach would be to randomly assign values for gene-pairs. scikit-learn in Python has tools for imputation that could be played around with (see PySingleCellNet if you go that route). No matter what, if you decide to play around with imputing either gene expression or gene-pair values, make sure that you explore performance to understand how the number of genes missign in the query data impact AUPR, accuracy, etc
